### ISSUE TITLE
Show thanks received 

### ISSUE THREAD

**Issue by maskaravivek** [commons-app/apps-android-commons#2386]
**Summary:** 

I recently received my first "Thanks" for a photo uploaded to Commons. It would be great if there's some way for the user to see the "Thanks" that he has received. 

In the first cut, it would be nice to open the web page showing the "Thanks logs" filtered for that user. 

https://commons.wikimedia.org/wiki/Special:Log?type=thanks&user=&page=User%3AMaskaravivek&wpdate=&tagfilter=

**Comment by vanshikaarora** [comment]
@maskaravivek Currently on tapping the notification we are taken to the details of the image uploaded. Shouldn't we add a small `SEE YOUR THANKS LOG`  or such a thing to take the user to the Web page in a webview?

**Comment by maskaravivek** [comment]
Notifications will show up only when it's unread. I was hoping to have a permanent option. 

Maybe we can add it on tap of "Thanks received" in Achievements activity.

**Comment by vanshikaarora** [comment]
>Maybe we can add it on tap of "Thanks received" in Achievements activity. 

Yes that's definitely a good option we can take the user to a separate activity for each of their Thanks Received

Or even we can also add this for each individual image. In thetoolbar for image details activity we can add an icon about the thanks received for that contribution

**Comment by sp2710** [comment]
>Maybe we can add it on tap of "Thanks received" in Achievements activity.

Maybe on clicking "Thanks received" we can redirect the user on a separate activity containing List of all the thanks received and on clicking on any item in the list the user should be redirected to the details of the image uploaded

**Comment by nicolas-raoul** [comment]
@maskaravivek  Could you please post a screenshot showing how the thank looks like, in your https://commons.wikimedia.org/wiki/Special:Notifications ? Thanks!

**Comment by maskaravivek** [comment]
<img width="707" alt="screenshot 2019-02-04 at 12 34 47 pm" src="https://user-images.githubusercontent.com/3069373/52194314-4bd64c00-2879-11e9-9650-2e14e24b2831.png">

**Comment by vanshikaarora** [comment]
>Does the URL in the issue description also allow to see what picture it was? Apparently not? Might be for privacy reasons, since thanks are public?

@nicolas-raoul Looking at the API available here:
https://commons.wikimedia.org/wiki/Special:ApiSandbox#action=thank&revid=456&source=diff&token=123ABC I can guess that currently the image url is not provided in the API request

but I guess it can be done using 
```
source

diff

A short string describing the source of the request, for example diff or history.
```

Please correct me if I am wrong here :)

**Comment by maskaravivek** [comment]
According to the documentation: 

> The specific revision that is being "thanked" for is not stored in the public logging table. Only the timestamp, sender, and recipient, are publicly logged.

The same can be checked in the database. Any information about the image being thanked is not persisted in the public DB. 

https://quarry.wmflabs.org/query/31756

At best, we can simply show a log of thanks received by the user. Am not sure if anyone would be interested in seeing that log without knowing which media was thanked.

**Comment by vanshikaarora** [comment]
>The specific revision that is being "thanked" for is not stored in the public logging table. Only the timestamp, sender, and recipient, are publicly logged.

- Presently the app shows the number of thanks received so showing the logs of that "Thanks recieved" might add an additional value for the user  
- Also (This is an optional suggestion) : since we have the timestamp we might be able to retrieve which image we are thanked for and thus give information of the image being thanked

**Comment by sp2710** [comment]
>The specific revision that is being "thanked" for is not stored in the public logging table. Only the timestamp, sender, and recipient, are publicly logged.

Is this the timestamp at which the thanks was received and not the timestamp at which the image was uploaded? In that case it might not be possible to retrieve which image we are thanked for.

**Comment by maskaravivek** [comment]
> * Presently the app shows the number of thanks received so showing the logs of that "Thanks recieved" might add an additional value for the user

I am fine with showing the logs but lets other pitch in too. cc @misaochan @neslihanturan @nicolas-raoul 
If we go ahead with it, 
- either we can simply open this link: https://commons.wikimedia.org/wiki/Special:Log?type=thanks&user=&page=User%3AMaskaravivek&wpdate=&tagfilter=

- or we can build a native page. For native page, we will need an API that returns the logs. 

> * Also (This is an optional suggestion) : since we have the timestamp we might be able to retrieve which image we are thanked for and thus give information of the image being thanked

Yes, as @sp2710 pointed out, this timestamp is for the thanks and not for image upload. So it might not be possible to link it with an image.
